src/client/Microsoft.Identity.Client/TokenCacheNotificationArgs.cs

```diff
@@ -43,5 +45,12 @@ public sealed partial class TokenCacheNotificationArgs
         /// Not all cache operations modify the state of the cache.
         /// </summary>
         public bool HasStateChanged { get; internal set; }
+
+        /// <summary>
+        /// Indicates wheather the cache change occured in the <see cref="ClientApplicationBase.UserTokenCache" />
+        /// or in the <see cref="ConfidentialClientApplication.AppTokenCache"/>. The latter is used in Client Credential grant, 
+        /// i.e. <see cref="ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{string})"/>
+        /// </summary>
+        public bool IsApplicationTokenCache { get; }
```

Comment:
Alternatively, we could invert the flag, expose `IsUserTokenCache` if you think it's more intuitive.

Reply:
this looks good to me.